Web_Development/Javascript/Topics/5.Syntax/2.Loops/README.md

```diff
@@ -0,0 +1,237 @@
+console.log('Hello World!');
+console.log('Hello World!');
+console.log('Hello World!');
+console.log('Hello World!');
+console.log('Hello World!');
+```
+```js
+// You can write this :
+for(var i = 1; i<=5; i++){
+    console.log('Hello World');
+}
+```
+## Different types of loops :
+- ``for`` - It loops through a block of code required number of times
+- ``for/in`` - It loops through the properties of an object
+- ``for/of`` - It loops through the values of an iterable object
+- ``while``- It loops through a block of code until a specific condition has not falsified
+- ``do/while`` - It also loops a block of code until a specific condition has not falsified
+
+### 1. For Loop:
+Syntax of ***for*** loop :
+```js
+for(intialization; condition; updation){
+    //Block of code
+}
+```
+- *Initialization* - This statement will execute only at the start of the loop
+- *Condition* - This statement will set a condition which will be checked before the start of every loop and the loop will run until it gets falsified
+- *Updation* - This statement will be executed after the completion of every loop
+
+***For Example :***
+```js
+let sum = 0;
+//Sum of first 100 natural numbers :
+for(var i = 1; i<=100; i++){
+    sum += i;
+}
+console.log("The sum of first 100 natural numbers : " + sum);
+```
+Output :
+```
+The sum of first 100 natural numbers : 5050
+```
+>### ***Curious Corner :***
+>
+>**Loop Scope** : 
+>Using ``var`` in a loop and ``let`` in a loop may give different results outside the loop
+>
+>***For Example :***
+>```js
+>var i = 10;
+>
+>for (var i = 0; i < 20; i++) {
+>  // block of code
+>}
+>// Here i is 20
+>```
+>```js
+>let i = 10;
+>
+>for (let i = 0; i < 20; i++) {
+>  // block of code
+>}
+>// Here i is 10
+>```
+> The reason is ``var`` re-declares the variable outside the scope of loop, this is alos known as ***Hoisting in JS***. But ``let`` do not declares the variable outside the loop and thus the scope of ``let`` is limited within the paranthesis of the loop 
+
+### 2. For In Loop:
+
+Syntax of ***for in*** loop :
+```js
+for(key in object){
+    //Block of code
+}
+```
+- *key* - You can imagine it as the index value of each element in an array. But in case of objects, we have key-value pairs associated with every entry in the object. Thus we can iterate over every entry of an object using it's keys.
+
+***For Example :***
+
+- For in over **objects**
+```js
+const student = {name:"Gurpreet", roll_no:"101", age:19};
+
+for (let key in student) {
+  console.log("Student "+key+" is "+student[key]);
+}
+```
+Output :
+```
+Student name is Gurpreet
+Student roll_no is 101
+Student age is 19
+```
+
+- For in over **arrays**
+```js
+const arr = [1,2,3,4,5];
+let sum = 0;
+
+for (let index in arr) {
+  sum += arr[index];
+}
+
+console.log("The array sum is : " + sum);
+```
+Output :
+```
+The array sum is : 15
+```
+
+>### ***Curious Corner :***
+>- Do not use ``for in`` over an array if the index **order** is important.
```

Comment:
Also write its proof, if you are claiming something

Reply:
Done sir :)